### PR TITLE
Aerospike collector support

### PR DESCRIPTION
**Pull request by travisgroth**
Title says it all.  Collect common stats from aerospike nodes.

### CODE REVIEW

File: src/collectors/aerospike/aerospike.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# coding=utf-8
+
+"""
+Collect statistics from Aerospike
+
+#### Dependencies
+
+ * socket
+ * telnetlib
```
Comment (shortdudey123) [inline src/collectors/aerospike/aerospike.py]: Any reason for using telnet as opposed to the aerospike python library?

**Comment by shortdudey123** [comment]
Can you fix the pep8 issues and generate the doc file?

**Comment by travisgroth** [comment]
Re: pep8 - do you have your pep8 config published?  I’m not getting those errors locally or in an online tool for some reason (like http://pep8online.com/checkresult <http://pep8online.com/checkresult>) so I can’t verify I’ve fixed them without some merge spam.

Re: will work on the doc file.  I’m seeing some issues with the types that are being generated in the docs.  How is that determined?  I’m getting tuples where I should be getting bools.

File: src/collectors/aerospike/aerospike.py (hunk 1 above)
Comment (travisgroth) [inline src/collectors/aerospike/aerospike.py]: Using a core module felt easier to manage from a dependencies perspective (rather than having to deploy the python and C client on every server).  

I don’t think the ‘info’ API in the python client library is actually any cleaner.  It appears to return string blobs that are about the same as we get back from the info port.

Keep in mind I’m not bypassing or re-implementing the client here; I’m explicitly using the plaintext port.

**Comment by shortdudey123** [comment]
`http://pep8online.com/` shows several errors as well
you can run pep8 on your code with `pep8 --config=.pep8 src/collectors/aerospike `

Doc is generated by running 
`python build_doc.py`
`python test.py`

**Comment by travisgroth** [comment]
> On Jul 5, 2016, at 8:28 PM, Grant Ridder <notifications@github.com> wrote:
> 
> http://pep8online.com/ shows several errors as well
> you can run pep8 on your code with pep8 --config=.pep8 src/collectors/aerospike
> 
https://travis-ci.org/python-diamond/Diamond/jobs/142610506 <https://travis-ci.org/python-diamond/Diamond/jobs/142610506>

Sorry, I read the CI output in a rush.  I cleaned up the E201/202/501 errors.  The rest I cannot reproduce locally with `pep8` or on pep8online.  Those are the ones I was asking about your config for.  Since I can’t test it I’m not positive what the issue is.  Here’s the latest build failure:

https://travis-ci.org/python-diamond/Diamond/jobs/142659358 <https://travis-ci.org/python-diamond/Diamond/jobs/142659358>

Thoughts?

> Doc is generated by running 
> python build_doc.py
> python test.py
> 

Got that sorted and found why I had strange option types.

**Comment by shortdudey123** [comment]
the pep8 config can be found at `.pep8` (this is the config that you see passed in the testing)

Verify you are using the newest version of the pep8 library (i am running several versions older on my laptop and am able to see all the same errors as travis

```
$ pep8 --version
1.5.7
$
$ pep8 --config=.pep8 src/collectors/aerospike/
src/collectors/aerospike/test/testaerospike.py:76:33: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:90:30: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:98:39: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:123:33: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:137:30: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:145:39: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:164:33: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:178:30: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:180:31: E121 continuation line under-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:190:39: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:200:43: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:201:44: E121 continuation line under-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:225:33: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:235:30: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:237:31: E121 continuation line under-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:246:39: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:257:43: E126 continuation line over-indented for hanging indent
src/collectors/aerospike/test/testaerospike.py:258:44: E121 continuation line under-indented for hanging indent
$ 
```

File: src/collectors/nginx/nginx.py [hunk 2]
```diff
@@ -1,35 +1,65 @@
 # coding=utf-8
```
Comment (shortdudey123) [inline src/collectors/nginx/nginx.py]: revert this file since it is unrelated

File: src/collectors/nginx/test/fixtures/plus_status [hunk 3]
```diff
@@ -0,0 +1,194 @@
+{
```
Comment (shortdudey123) [inline src/collectors/nginx/test/fixtures/plus_status]: revert this file since it is unrelated

**Comment by shortdudey123** [comment]
looks like you got the nginx collector mixed up in this PR

**Comment by travisgroth** [comment]
Hrm.  I’m on pep8 1.7.0.

$ pep8 --version
1.7.0
[feature_aerospike]
$ pep8 --config .pep8 src/collectors/aerospike/
[feature_aerospike]
$

**Comment by travisgroth** [comment]
I'm failing at github this evening.  Going to open a new PR with a clean branch as it doesn't seem like an FP allows me to re-open this PR without the bad merge history.  Sorry about the noise.
